samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -769,6 +768,27 @@ class SamzaContainer(
       else
         Thread.sleep(Long.MaxValue)
     } catch {
+      case e: InterruptedException =>
```

Comment:
Yes that is possible even without this change when stream processor attempts to shutdown the container, times out and then forcefully shuts container executor down by invoking `shutdownNow`.  We should handle that in a separate PR since its useful to clean up components as much as possible. However, this PR doesn't introduce any additional interrupt scenarios after shutdown is triggered.